packages/flutter/test/widgets/clip_test.dart

```diff
@@ -358,7 +358,7 @@ void main() {
       find.byType(RepaintBoundary).first,
       matchesGoldenFile('clip.ClipRect.1.png'),
     );
-  });
+  }, tags: 'golden');
```

Comment:
I'm not sure we can implement this on the Web due to browser restrictions. Possible alternatives to tags:

- Make `matchesGoldenFile` always succeed on the Web (the downside is users may incorrectly assume their tests are passing when they actually fail).
- Create a `testGolden` that's a sibling of `testWidgets`. `testGolden` only runs in `flutter_tester` and never on the Web.

Reply:
cc @Piinks who is working on redoing how goldens work

Reply:
There is a way to do golden testing on the web, but it might require a slightly different API. Its possible to ask chrome via devtools (on the server side) for a screenshot of particular dimensions. 

We would need to move the comparison to the backend and use package:tests's communication channel to forward to result of the comparison. I can write up a small one-pager on what this looks like

Reply:
I meant we may not be able to take a screenshot of a widget. But now that I looked at Flutter's implementation, all it does is create a standalone `Scene` and takes a screenshot of that. Perhaps we could do the same by rendering the scene into a separate iframe.

Reply:
You can ask chrome (from the serving side) for a screen shot at particular dimensions, and we can get those from the layer tree presumably.

Reply:
May be a minor distinction, but if one layer appears on top of another layer, will Chrome isolate the layer that's being screenshotted? That's what Flutter does.

Reply:
ahh good point, I'm not sure - though I doubt our concept of a layer lines up nicely with Chrome's even if it was supported.

Reply:
If you can capture a screenshot, you can test it using the gold process I've been working on. There's a number of ways you can differentiate web screenshots from others from implementing unique naming conventions to distinguish platform-specific tests or a unique config/key:value pair for a given test. I have a design doc laying out the possible workflows with gold.